modules/default/compliments/compliments.js

```diff
@@ -63,20 +63,28 @@ Module.register("compliments", {
 				self.config.compliments = JSON.parse(response);
 				self.updateDom();
 			});
-		} else if (this.config.advice) {
-			var xobj = new XMLHttpRequest();
-			xobj.overrideMimeType("application/json");
-			xobj.open("GET", "https://api.adviceslip.com/advice", true);
-			xobj.onreadystatechange = function() {
-				if (xobj.readyState === 4 && xobj.status === 200) {
-					const adviceResp = JSON.parse(xobj.responseText);
-					self.config.compliments = adviceResp.slip.advice
-					self.updateDom();
-				}
-			};
-			xobj.send(null);
 		}
 
+    if (this.config.advice) {
+        var xobj = new XMLHttpRequest();
+        xobj.overrideMimeType("application/json");
+        xobj.open("GET", "https://api.adviceslip.com/advice/search/a", true);
```

Comment:
maybe make the searched letter a randon one?

Reply:
Yeah we could, searching for 'a' seems to return the most possible results after my admittedly short testing. With this 'a' search we get 192 results, which i then filter for results less than 35 chars.

Reply:
Looking at the advice api I see they also offer a random adivce call. How about not loading a set of advices here but rather call https://api.adviceslip.com/#endpoint-random in the randomCompliment function?